Tasks/Common/azure-arm-rest/azure-arm-resource-filter.ts

```diff
@@ -0,0 +1,76 @@
+import msRestAzure = require("./azure-arm-common");
+import tl = require('vsts-task-lib/task');
+import util = require("util");
+import webClient = require("./webClient");
+import azureServiceClient = require("./AzureServiceClient");
+import Model = require("./azureModels");
+import Q = require("q");
+
+export class AzureResourceFilterManagementClient extends azureServiceClient.ServiceClient {
+    public _azureResourceFilter: AzureResourceFilter;
+
+    constructor(credentials: msRestAzure.ApplicationTokenCredentials, subscriptionId: string, options?: any) {
+        super(credentials, subscriptionId);
+
+        if (!!options && !!options.longRunningOperationRetryTimeout) {
+            this.longRunningOperationRetryTimeout = options.longRunningOperationRetryTimeout;
+        }
+        this._azureResourceFilter = new AzureResourceFilter(this);
+    }
+}
+
+export class AzureResourceFilter {
+    private _client: AzureResourceFilterManagementClient;
+
+    constructor(client: AzureResourceFilterManagementClient) {
+        this._client = client;
+    }
+
+    public getResources(resourceType: string, resourceName: string, options: any, callback: azureServiceClient.ApiCallback) {
```

Comment:
while the callback based approach is what is used most in this library - for newer methods we are keeping it all in promises for cleaner code. If you want to use Promises for the public methods - that might keep the code simpler. This method would be changed to
public getResources(resourceType: string, resourceName: string) : Promise<any>

Reply:
Created a wrapper class that resolves this callback as a promise.
https://github.com/Microsoft/vsts-tasks/pull/6001/files#diff-6bc02ddedf82bfe34fa4693469e1736eR20

Reply:
Don't have the callback version at all. That will help you avoid creating the wrapper

Reply:
Removed callback as discussed offline.